RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitWorkerRecover.cc

```diff
@@ -175,9 +177,10 @@ bool EcalRecHitWorkerRecover::run(const edm::Event& evt,
     EcalTrigPrimDigiCollection::const_iterator tp = tpDigis->find(ttDetId);
     // recover the whole trigger tower
     if (tp != tpDigis->end()) {
+      EcalTPGScale ecalScale{ecalScaleTokens_, *eventSetup_};
```

Comment:
is this really fast? The implementation suggests 3 ```EventSetup::getData``` calls.
This piece is inside of a call that can repeat many times per event.
Better keep these cached in the ```::set``` method (same comment applies to the other case in this file)

Reply:
@slava77 this is likely faster than the old code and at worst only just as fast. In the old code, each call to functions of `EcalTPGScale` were making the 3 calls to `EventSetup::getData`. Now we only do it in the constructor and cache the results so they can be reused.

Reply:
I'm confused, ```ecalScale``` is clearly not cached in this case/loc.
Wouldn't it be faster/more effective to cache ecalScale in the ::set method (where a few other ES data is cached already)?

Reply:
In the old code, caching `ecalScale` was only caching a pointer to the `EventSetup`, which the new code also does. In the old code, each call to a function of `EcalTPGScale` would call multiple `EventSetup::getData` calls. Now this is only done in the constructor. So the changes we made are better.

Caching ecalScale would be more difficult since `EcalTPGScale` now requires the `EventSetup` to be passed to its constructor (since all other uses of `EcalTPGScale` other than in `EcalRecHitWorkerRecover` were setting it right after calling the constructor) which means `EcalRecHitWorkerRecover` could not hold an `EalTPGScale` but would have to hold a `std::unique_ptr`. As I'm dealing with an undergraduate and our only purpose is to get the `esConsumes` done and the fact that the changes we made here are already a performance improvement over the old code, I'd prefer not to do any further changes in this pull request. If further changes are wished, I'd prefer a GitHub issue be made and the actual developers of this code do futher changes.

Reply:
> Caching ecalScale would be more difficult since `EcalTPGScale` now requires the `EventSetup` to be passed to its constructor

It's unclear to me why adding ```ecalScale_ =  EcalTPGScale ecalScale(ecalScaleTokens_, es)``` in ```EcalRecHitWorkerRecover::set(const edm::EventSetup& es)``` is complicated.
Perhaps I'm missing something (was e.g. a default constructor or an assignment operator removed).

Reply:
There is no default constructor